### PR TITLE
Issue/256 consensus techdebt unexport fields

### LINKED ISSUE #256: [TECHDEBT][Consensus] Do not export any fields in the ConsensusModule struct implementation
## Objective

Do not export any fields in the `ConsensusModule` struct implementation.

## Origin Document

During the review process of #250, the following discussion came up in [this thread](https://github.com/pokt-network/pocket/pull/250#discussion_r982346486).

The reason for this is due to the tests in `pacemaker_test.go`:

```go
	for _, pocketNode := range pocketNodes {
		// utilityContext is only set on new rounds, which is skipped in this test
		utilityContext, err := pocketNode.GetBus().GetUtilityModule().NewContext(int64(testHeight))
		require.NoError(t, err)

		consensusModImpl := GetConsensusModImplementation(pocketNode)
		consensusModImpl.FieldByName("Height").SetUint(testHeight)
		consensusModImpl.FieldByName("Step").SetInt(testStep)
		consensusModImpl.FieldByName("LeaderId").Set(reflect.Zero(reflect.TypeOf(&leaderId))) // This is re-elected during paceMaker catchup
		consensusModImpl.FieldByName("UtilityContext").Set(reflect.ValueOf(utilityContext))
	}
```

As of PR #250, the only fields that remain exported are:
- Height
- Round
- Step
- Block
- UtilityContext

## Goals

- Make all fields in the `ConsensusModule struct lowercased (i.e. internal to the consensus package)
- Remove the dependency on reflections in pacemaker unit tests

## Deliverable

- [ ] A Pull Request that meets the goals above

## Non-goals / Non-deliverables

- Developing a new type of testing framework
- Modifying any consensus-related business logic

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## Testing Methodology

- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @Olshansk 
**Co-Owners**: ???

### PR DESCRIPTION
**Pull request by gokutheengineer**
## Description

Remove unexported fields in the consensus module implementation.

## Issue

Fixes #256 

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [X] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Unexpected exported fields of the `ConsensusModule` used by the `pacemaker_test.go`
- Created a `ConsensusDebugModule` interface
- Wrote setter methods for the unexpected fields under the `ConsensusDebugModule` interface
- Refactored code where the unexported fields are mentioned

## Testing

- [X] `make test_consensus`
- [X] `make develop_test`
- [X] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [X] I have tested my changes using the available tooling
- [x] I have updated the corresponding CHANGELOG

### If Applicable Checklist
- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: consensus/module.go [hunk 1]
```diff
@@ -76,6 +76,40 @@ type consensusModule struct {
 	messagePool map[typesCons.HotstuffStep][]*typesCons.HotstuffMessage
 }
 
+// Functions exposed by the debug interface should only be used for testing puposes.
+type ConsensusDebugModule interface {
```
Comment (Olshansk) [inline consensus/module.go:80]: @gokutheengineer Can you add a TODO next to `HandleDebugMessage` in `shared/modules/consensus_module.go` to move it into a `Debug` module in the future?

@deblasis Check out this pattern Gohkan put together. For functions that we want to expose for testing / debugging, but keep it explicitly clear (to the developers)  that it's not part of  as not part of the main interface, we create this Debug module. I'm a fan

**Review by Olshansk — CHANGES_REQUESTED** [review]
Pretty clean and smooth change. Aside from he request to add that one TODO comment, can you also just update `consensus/doc/CHANGELOG.md`? After that, good to go

**Review by deblasis — APPROVED** [review]
LGTM, very nice one @gokutheengineer !

**Comment by gokutheengineer** [comment]
@Olshansk comments addressed, let me know if there are some other things to address

File: consensus/doc/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.0.12] - 2022-12-12
+- Unexport `ConsensusModule` fields
+- Create `ConsensusDebugModule` interface with setter functions to be used only for debugging puroposes
+- Update test in `TestPacemakerCatchupSameStepDifferentRounds` in pacemaker_test.go to use setter functions
```
Comment (Olshansk) [inline consensus/doc/CHANGELOG.md:13]:
```suggestion
- Update test in `TestPacemakerCatchupSameStepDifferentRounds` in `pacemaker_test.go` to use setter functions
```

**Review by Olshansk — APPROVED** [review]
One small in place suggestion but otherwise LGTM!

In terms of merging, what we've been doing is a "squash and merge" where the body of the commit is the PR description.

**Pull request merged**